ysh/val_ops.py

```diff
@@ -120,6 +120,11 @@ def Stringify(val, blame_loc, prefix=''):
             val = cast(value.Eggex, UP_val)
             s = regex_translate.AsPosixEre(val)  # lazily converts to ERE
 
+        elif case(value_e.List):
+            raise error.TypeErrVerbose(
+                "%sexpected Null, Bool, Int, Float, Eggex, got List. Use '@' instead of '$' to splice the returned List into Strings" % prefix,
```

Comment:
How about something shorter, which still conveys the point (users tend not to read long messages):

- "Use @ instead of $ to splice" ?
- "Use @ instead of $ to splice a List"
- "List can't be substituted.  Use @ instead of $ to splice"

I kind of like the last one?

Reply:
Hmm yeah my first idea is very long actually! I was also thinking about displaying a simple help command to show more infos. I just checked if there is any reasonable doc page to reference, but I couldn't find one. Is there some (reference) docs about switching between command & expression mode? Then we could have a message like
`Use @ instead of $ to splice or see "help switch-modes"` or something like that.